docs/index.rst

```diff
@@ -8,6 +8,8 @@ Contents:
 .. toctree::
    :maxdepth: 3
 
-   jdbc_connector
-   configuration_options
+   source_connector
```

Comment:
Does adding hierarchy here make sense (i.e. Source connector -> (description, configs))? Or is that just not warranted for what we have so far? Just curious because the interleaved ordering seems a bit odd.